### ISSUE TITLE
Parse Filter Schema Definitions to React Filter Panel Components

### ISSUE THREAD

**Issue by bryanculver** [nautobot/nautobot#3583]
With #3983 @norbert-mieczkowski-codilime added the filtering side panel to the new UI.

With #4162 @timizuoebideri1 extended the Dynamic Filter form to fully emit JSON representation of the fields, lookup types, and field definitions.

In this story should preset in the filtering side panel the choice dropdown of:

- Which field would you like to filter on
- Which lookup expression would you like to choose that's available for the field chosen above
- Present an input box to enter values to filter on

We will create a separate story to map the output of #4162 to each different field type (multi-choice select, dropdown, text input similar to #2381)

---
Carved from #2806, Related (and blocked by): #3582

**Context:**
- [ ] The model field to filter field mapping at `BaseFilterSet.FILTER_DEFAULTS` needs to also include the correct UI widget types in the `extra` for the filter mapping
    - [x] Prototype a JSON representation of the FilterSet to see if we can get away of needing widgets at all for the new UI. JSON Schema would be a good place to start/model from
    - [x] Serve the above representation on the UI API by some convention to the react UI, Ask for a model and get it's filters and their classes and args. We likely will need to advance it to allow for plugin views which define their own filterset (so maybe the view may need to hint that it needs a different filterset than the default one by the model).
    - [ ] Build the filterset => widget logic in the React UI, similar to what has been done here: https://github.com/nautobot/nautobot/blob/0fffa8ee6d4168fdb11b27019e500e0e1ef9fc1c/nautobot/utilities/utils.py#L866, but in React

**This Story:**

Addresses that last point which is taking the schema and rendering the components it in the UI.

**Comment by bryanculver** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Nautobot-Core-Engineering-64a875e74842500d9e80e23e/issues/nautobot/nautobot/3583?planning-poker) with Zenhub @jathanism

**Comment by bryanculver** [comment]
@norbert-mieczkowski-codilime for the final input field I think it would be valuable to start with
- https://github.com/csandman/chakra-react-select
- https://github.com/JedWatson/react-select

**Comment by norbert-mieczkowski-codilime** [comment]
Yeah, I already implemented `react-select` in Nautobot UI https://github.com/nautobot/nautobot-ui/tree/develop/src/components/ReactSelect. Regarding `chakra-react-select`, the library history shows that it has rather rare maintenance cycles and inactive community, so I decided not to use it.

**Comment by bryanculver** [comment]
We have https://github.com/nautobot/nautobot/issues/4462 which will take this the rest of the way. Implemented a majority of this in #4434
